scipy/linalg/tests/test_matfuncs.py

```diff
@@ -838,3 +838,23 @@ def test_expm_cond_fuzz(self):
             # eps times the condition number kappa.
             # In the limit as eps approaches zero it should never be greater.
             assert_array_less(p_best_relerr, (1 + 2*eps) * eps * kappa)
+
+
+class TestKhatriRao(object):
+
+    def test_basic(self):
+        a = khatri_rao(array([[1, 2], [3, 4]]),
+                       array([[1, 2], [3, 4]]))
+
+        assert_array_equal(a, array([[1, 4],
+                                     [3, 8],
+                                     [3, 8],
+                                     [9, 16]]))
+
```

Comment:
Could you add tests here for other unhandled shapes as well, e.g. 0/1/3-D arrays?

And a separate test for empty arrays would be good - they should work and give back a correctly-shaped empty array I think.

Reply:
ok

Reply:
Still missing the functionality of what happens if any of the arrays are empty.